modules/visual_script/visual_script.cpp

```diff
@@ -291,80 +292,113 @@ void VisualScript::_node_ports_changed(int p_id) {
 #endif
 }
 
-void VisualScript::add_node(const StringName &p_func, int p_id, const Ref<VisualScriptNode> &p_node, const Point2 &p_pos) {
+void VisualScript::add_node(const StringName &p_name, int p_id, const Ref<VisualScriptNode> &p_node, const Point2 &p_pos) {
 	ERR_FAIL_COND(instances.size());
-	ERR_FAIL_COND(!functions.has(p_func));
 	ERR_FAIL_COND(p_node.is_null());
 
-	for (Map<StringName, Function>::Element *E = functions.front(); E; E = E->next()) {
-		ERR_FAIL_COND(E->get().nodes.has(p_id)); //id can exist only one in script, even for different functions
+	StringName var_name;
+	if (Object::cast_to<VisualScriptVariableGet>(*p_node)) {
+		Ref<VisualScriptVariableGet> vget = p_node;
+		var_name = vget->get_variable();
+	} else if (Object::cast_to<VisualScriptVariableSet>(*p_node)) {
+		Ref<VisualScriptVariableSet> vset = p_node;
+		var_name = vset->get_variable();
 	}
+	if (variables.has(var_name)) {
+		for (Map<StringName, Variable>::Element *E = variables.front(); E; E = E->next()) {
+			ERR_FAIL_COND(E->get().nodes.has(p_id)); //id can exist only one in script, even for different variables
+		}
 
-	Function &func = functions[p_func];
+		Variable &var = variables[var_name];
+		var.nodes[p_id] = p_node;
+	}
 
-	if (Object::cast_to<VisualScriptFunction>(*p_node)) {
-		//the function indeed
-		ERR_FAIL_COND_MSG(func.function_id >= 0, "A function node has already been set here.");
+	if (functions.has(p_name)) {
+		for (Map<StringName, Function>::Element *E = functions.front(); E; E = E->next()) {
+			ERR_FAIL_COND(E->get().nodes.has(p_id)); //id can exist only one in script, even for different functions
+		}
 
-		func.function_id = p_id;
-	}
+		Function &func = functions[p_name];
+		if (Object::cast_to<VisualScriptFunction>(*p_node)) {
+			//the function indeed
+			ERR_FAIL_COND_MSG(func.function_id >= 0, "A function node has already been set here.");
 
-	Function::NodeData nd;
-	nd.node = p_node;
-	nd.pos = p_pos;
+			func.function_id = p_id;
+		}
 
+		Function::NodeData nd;
+		nd.node = p_node;
+		nd.pos = p_pos;
+		func.nodes[p_id] = nd;
+	}
+ 
```

Comment:
There seems to be trailing whitespace on this line (select diff to see it): https://github.com/godotengine/godot/runs/5821764351?check_suite_focus=true

Reply:
Oops! I forgot to format before pushing :)